en/docs/learn/api-security/oauth2/grant-types/jwt-grant.md

```diff
@@ -1,69 +1,137 @@
+            
+            Since we have used WSO2 IS as the identity provider here, follow the below steps to create the identity provider certificate from the `wso2carbon.jks file`.
+
+            - Open your command line interface, navigate to the `<IS_HOME>/repository/resources/security/` directory and run the following command.
+              ``` java
+              keytool -export -alias wso2carbon -file wso2.crt -keystore wso2carbon.jks -storepass wso2carbon
+              ```
 
-7.  Enter a **Callback URL.** For example, use `https://wso2is.local:8080/playground2/oauth2client` and click **Add**. The **OAuth Client Key** and **OAuth Client Secret** will now be visible.
+            - Once you run this command, the `wso2.crt` file will be generated inside the same folder. You can upload it by clicking **Choose File** in Identity Provider section in the Management Console.
+
+            See [Using Asymmetric Encryption](https://docs.wso2.com/display/ADMIN44x/Using+Asymmetric+Encryption) in the WSO2 Product Administration Guide for more information on how public keys work and how to sign these keys by a certification authority.
+
+    -   **Alias** : Give the audience (`aud` value) of the JWTs issued by the identity provider (WSO2 IS).
+
+    !!! tip
+         When testing, you can find the values such as issuer name (`iss`) and audience (`aud`) by decoding the JWT obtained in [Step 1](#step-1-obtain-a-jwt-from-an-external-identity-provider) using a JWT decoder such as [https://jwt.io](https://jwt.io). For more information, see [JWT Bearer Grant](#jwt-bearer-grant).
+
+    [![]({{base_path}}/assets/img/learn/api-security/oauth2/jwt-grant/add-identity-provider.png)]({{base_path}}/assets/img/learn/api-security/oauth2/jwt-grant/add-identity-provider.png)
+
+4. Click **Register** to complete the registration of the identity provider.
+
+5. Navigate to the **Service Providers** section in the **Main** -> **Identity** section and click **Add.**
+
+6. Fill in the **Service Provider Name** and provide a brief **Description** of the service provider. See [Adding a Service Provider](https://is.docs.wso2.com/en/5.10.0/learn/adding-and-configuring-a-service-provider/) for more information.
+
+6. Expand the **OAuth/OpenID Connect Configuration** under Inbound Authentication Configuration and click **Configure.**
+
+7. Enter a **Callback URL.** For example, use `http://localhost:8080/playground` and click **Add**. The **OAuth Client Key** and **OAuth Client Secret** will now be visible.
 
     [![]({{base_path}}/assets/img/learn/add-service-provider.png)]({{base_path}}/assets/img/learn/add-service-provider.png)
 
-### Using the JWT grant
+Now we have a registered identity provider and as well as a service provider with an **OAuth Client Key** and an **OAuth Client Secret**. In the next section we can test the JWT grant with the configured setup.
+
+## Using the JWT grant
```

Comment:
The same topic repeated twice.

Reply:
Restructured the steps via https://github.com/wso2/docs-apim/pull/3170/commits/938074070570a12605876265c41b0f28362e63d2